common/include/pcl/common/random.h

```diff
@@ -68,6 +68,38 @@ namespace pcl
       using type = std::normal_distribution<T>;
     };
 
+    template <typename RandomEngineT = std::default_random_engine>
+    class RandomBase {
+    public:
+      using SeedT = typename RandomEngineT::result_type;
+
+      RandomBase(SeedT seed = std::random_device()()) : seed_(seed), rng_(seed) {}
+
+      inline SeedT
+      getSeed() const
+      {
+        return seed_;
+      }
+
```

Comment:
```suggestion
inline void newSeed() 
{
  setSeed(std::random_device()());
}
```

This might be useful since I guess most people will be using this for reseeding.

Reply:
I would wait to see more code making use of the class before adding it. It's just replacing a single line of code.

Reply:
This feels more like a utility function for PCL users than PCL itself, though the default behavior of this class takes care of most of the use-cases.